go/cmd/api-gateway/requests.go

```diff
@@ -36,18 +40,35 @@ func requestHandler() http.HandlerFunc {
 			return
 		}
 
+		userPb := &pb.User{
+			Id:       reqApproval.User.Id,
+			UserName: reqApproval.User.UserName,
+		}
+
 		// Convert the JSON request to a protobuf message
 		protoRequest := &pb.RequestApproval{
-			Type: reqApproval.Type,
-			User: &pb.User{
-				Id:       reqApproval.User.Id,
-				UserName: reqApproval.User.UserName,
-			},
+			Type:             reqApproval.Type,
+			User:             userPb,
 			DataProviders:    reqApproval.DataProviders,
-			SyncServices:     reqApproval.SyncServices,
 			DestinationQueue: "policyEnforcer-in",
 		}
 
+		// Unmarshal JSON into a regular Go struct
+		var bodyJsonObj map[string]interface{}
+		if err := json.Unmarshal(body, &bodyJsonObj); err != nil {
+			logger.Sugar().Errorf("Error unmarhsalling get request: %v", err)
+			return
+		}
+
+		dataRequest, err := prepareDataRequestStruct(protoRequest.Type, bodyJsonObj, userPb)
+		if err != nil {
+			logger.Sugar().Errorf("Error preparing data request: %v", err)
+			return
+		}
+		protoRequest.Options = dataRequest.Options
+
+		logger.Sugar().Debugf("Data Request of type %s prepared", dataRequest.Type)
+
```

Comment:
the 
go func() {
			_, err := c.SendRequestApproval(ctx, protoRequest)
			
Should probably me switched around with creating the response channel to prevent a race condition. In that case the go func can also be removed.